src/add_caps.jl

```diff
@@ -113,17 +112,25 @@ end
         m = length(all_versions)
         # Bound the latest version
         version = all_versions[end]
-        add_caps!(compat,
-                  deps,
-                  UpperBound(),
-                  registry_paths,
-                  pkg_to_latest_version,
-                  pkg_to_latest_zero_version,
-                  pkg,
-                  pkg_path,
-                  version)
+        if option isa CapLatestVersion
+            add_caps!(compat,
+                    deps,
+                    UpperBound(),
+                    registry_paths,
+                    pkg_to_latest_version,
+                    pkg_to_latest_zero_version,
+                    pkg,
+                    pkg_path,
+                    version)
+        elseif !isa(option, ExcludeLatestVersion())
+            error("add_caps requires ExcludeLatestVersion or CapLatestVersion option")
+        end
         # Propagate the bounds backwards
         for j = m-1:-1:1
+            if !haskey(compat, version)
+                @warn "$compat_toml is suspicious"
```

Comment:
This happens when a package has no bounds at all for its latest version, but bounds for previous versions. Two packages in General had that...